addons/product/views/product_document_views.xml

```diff
@@ -19,6 +20,7 @@
                                required="type == 'binary'"
                                class="oe_inline"/>
                         <field name="url" widget="url" invisible="type == 'binary'" required="type == 'url'"/>
+                        <field name="res_name" string="Product Variant" invisible="res_model == 'product.template'"/>
```

Comment:
```suggestion
                        <field name="res_name" string="Product Variant" invisible="res_model != 'product.product'"/>
```
Might be exactly the same, but would be more logical IMHO, this is only a suggestion though, feel free to ignore